mkosi/__init__.py

```diff
@@ -529,13 +530,13 @@ def btrfs_subvol_delete(path: str) -> None:
         if not line:
             continue
         child_subvol_path = line.split(" ", 8)[-1]
-        child_path = os.path.normpath(os.path.join(path, cast(str, os.path.relpath(child_subvol_path, subvol_path))))
+        child_path = path / cast(str, os.path.relpath(child_subvol_path, subvol_path))
```

Comment:
I used a cast because it's a noop: `os.path.relpath` returns `str`, but for some reason mypy doesn't understand that. `Path()` would actually create a new object, which isn't necessary. I expect that we'll just drop the cast later on.